### PR TITLE
DEVX-2238: Update validation library to reflect install.yml changes

### PR DESCRIPTION
**Pull request by slitviachenko**
## Description
Updated validation logic to reflect recent install.yml changes
See https://wiki.corp.adobe.com/display/AdobeCloudPlatform/App+Builder+Template+Install+Configuration

## Related Issue
DEVX-2238

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by AmyJZhao — COMMENTED** [review]
Did you run lint on the branch?

**Comment by slitviachenko** [comment]
> Did you run lint on the branch?

@AmyJZhao ,
I've added 2 missing semicolons. Now lint doesn't complain.

**Review by AmyJZhao — APPROVED** [review]

File: src/checks/checkExtension.js [hunk 1]
```diff
@@ -11,15 +11,16 @@ governing permissions and limitations under the License.
 */
 
 /**
- * Check that extension are listed in install.yml file
+ * Check that if the "extension" property is specified in the install.yml file it is an object containing the "serviceCode" key.
```
Comment (roribio) [inline src/checks/checkExtension.js:14]:
```suggestion
 * Check that: if the "extension" property is specified in the install.yml file, it is an object containing the "serviceCode" key.
```

File: src/checks/checkExtension.js [hunk 2]
```diff
@@ -11,15 +11,16 @@ governing permissions and limitations under the License.
...
  * @returns {Promise<{message: string, status: string}>}
  */
-async function checkExtensions(fileData) {
-    if (!fileData.extension) {
+async function checkExtension(fileData) {
+    // "extension" is an optional property
+    if (fileData.extension && (fileData.extension.constructor.name !== 'Object' || !fileData.extension.serviceCode)) {
         return {
-            message: 'install.yml must specify extension points',
+            message: '"extension" must provide the "serviceCode" key.',
```
Comment (roribio) [inline src/checks/checkExtension.js:23]: Have you considered doing this validation using YAML Schema? For example, https://www.npmjs.com/package/yaml-schema-validator

**Review by roribio — APPROVED** [review]
Left a couple of suggestions, but this looks good to me 👍

File: src/checks/checkExtension.js (hunk 2 above)
Comment (slitviachenko) [reply src/checks/checkExtension.js:23]: @roribio , I am planning to do so. Right now the structure of [install.yml](https://wiki.corp.adobe.com/display/AdobeCloudPlatform/App+Builder+Template+Install+Configuration) file is not finalized yet. I am planning to re-use the yaml schema once it is finalized and available.

**Pull request merged**